grovel/grovel.lisp

```diff
@@ -802,8 +802,12 @@ string."
     (process-wrapper-form out form)))
 
 (define-wrapper-syntax in-package (name)
-  (setq *package* (find-package name))
-  (push `(in-package ,name) *lisp-forms*))
+  (let ((desired-package (find-package name)))
+    (assert desired-package ()
```

Comment:
Do you think it'd be worthwhile to have `(assert (find-package name) (name) ...` instead? That'd provide a restart for trying another package name. Probably overkill...